app/assets/javascripts/controllers/schedule/schedule_form_controller.js

```diff
@@ -381,9 +398,20 @@ ManageIQ.angular.app.controller('scheduleFormController', ['$http', '$scope', 's
     return $scope.scheduleModel.log_protocol === 'AWS S3';
   };
 
+  $scope.undefinedScheduleField = function(value) {
+    return (value === '' || typeof value === 'undefined');
```

Comment:
`typeof value === 'undefined'`  will never be `true` as `$scope.scheduleModel.log_aws_region` and `$scope.scheduleModel.security_protocol` are always initiated in `init` to be `""`.  I would use `! $scope.scheduleModel.log_aws_region` or `$scope.scheduleModel.log_aws_region === ""` instead of this function to have it more readable.

Reply:
@ZitaNemeckova this was added as part of the AWS S3 PR a while back and has nothing to do with the current PR, which is for support of OpenStack Swift.